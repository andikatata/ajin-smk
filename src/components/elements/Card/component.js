import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';

export default class Component extends React.Component {
  render() {
    const { children, classes, className } = this.props;
    const customClass = className ? `${classes.card} ${className}` : classes.card;

    return (
      <Card className={customClass}>
        {children}
      </Card>
    );
  }
}

Component.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.element,
  ]),
  classes: PropTypes.object,
  className: PropTypes.string,
};