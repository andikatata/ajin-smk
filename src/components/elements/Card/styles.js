const styles = {
  card: {
    minWidth: '200px',
    height: '310px',
    borderRadius: '3px',
    boxShadow: `0 2px 8px 0 rgba(0, 0, 0, 0.08)`,
    backgroundColor: '#ffffff',
  },
};

export default styles;