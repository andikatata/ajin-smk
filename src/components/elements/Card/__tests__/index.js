import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../index';

jest.mock('react-router-dom');

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Card className={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});