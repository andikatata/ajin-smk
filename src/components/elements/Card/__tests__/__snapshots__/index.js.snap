// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation1-6 MuiPaper-rounded-4 MuiCard-root-2 Component-card-1 [object Object]"
/>
`;
